### PR TITLE
Creating the page to show institutional links of an institution

### PR DESCRIPTION
**Pull request by tiagolimpe**
**Feature/Bug description:** Enabling option in the left menu of institution page to go to the institutional links page. And creating this page.

**Solution:** Adding a short controller to show the information of institutional links and making a view to this.

screenshot from 2018-04-16 16-27-23

screenshot from 2018-04-16 16-28-08

screenshot from 2018-04-16 16-28-24

screenshot from 2018-04-16 16-30-03

screenshot from 2018-04-17 15-10-03

**TODO/FIXME:** n/a

### CODE REVIEW

File: frontend/institution/institutional_links.html [hunk 1]
```diff
@@ -0,0 +1,54 @@
+<div layout="row" flex layout-align="center">
```
Comment (mayzabeel) [inline frontend/institution/institutional_links.html]: check the indentation of html

File: frontend/institution/institutionController.js [hunk 2]
```diff
@@ -469,4 +474,47 @@
         getFollowers();
 
     });
+
+    app.controller("InstitutionLinksController", function InstitutionLinksController($state, InstitutionService, MessageService) {
```
Comment (mayzabeel) [inline frontend/institution/institutionController.js]: test

File: frontend/institution/base_institution_page.html [hunk 3]
```diff
@@ -124,13 +124,11 @@
...
-                                        <md-menu-item ng-if="false">
-                                            <md-button ng-class="institutionCtrl.getSelectedItemClass('settings')"
-                                                ng-click="institutionCtrl.goToCommingSoon(institutionCtrl.institution.key)">
-                                                <md-icon>settings</md-icon>
-                                                <b>Vinculos Institucionais</b>
+                                        <md-menu-item>
+                                            <md-button ng-class="institutionCtrl.getSelectedItemClass('institutional_links')"
+                                                ng-click="institutionCtrl.goToLinks(institutionCtrl.institution.key)">
+                                                <md-icon>account_balance</md-icon>
+                                                <b>Vinculos institucionais</b>
```
Comment (raonismaneoto) [inline frontend/institution/base_institution_page.html]: Vínculos*

File: frontend/institution/institutionController.js [hunk 4]
```diff
@@ -189,6 +189,11 @@
             $state.go('app.institution.events', {institutionKey: institutionKey, posts: institutionCtrl.posts});
         };
 
+        institutionCtrl.goToLinks = function goToLinks(institutionKey) {
```
Comment (raonismaneoto) [inline frontend/institution/institutionController.js]: test it. Even this function being so small, the tests can protected it from changes.

File: frontend/institution/institutionController.js [hunk 5]
```diff
@@ -469,4 +474,47 @@
...
+            return !_.isEmpty(instLinksCtrl.childrenInstitutions);
+        };
+
+        function loadInstitution() {
+            InstitutionService.getInstitution(currentInstitutionKey).then(function success(response) {
+                instLinksCtrl.parentInstitution = response.data.parent_institution;
+                instLinksCtrl.childrenInstitutions = response.data.children_institutions;
+                instLinksCtrl.isLoadingInsts = false;
+            }, function error(response) {
+                $state.go("app.user.home");
```
Comment (raonismaneoto) [inline frontend/institution/institutionController.js]: Why are you sending him to home suddenly?

File: frontend/test/specs/institutionLinksControllerSpec.js [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+(describe('Test InstitutionLinksController', function() {
+
+    var instLinksCtrl, httpBackend, scope, institutionService, createCtrl, state;
+
+    var INSTITUTIONS_URI = "/api/institutions/";
+
+    var parent_institution_test = {
+        name: 'parentInst',
+        key: 'parentKey',
```
Comment (raonismaneoto) [inline frontend/test/specs/institutionLinksControllerSpec.js]: remove the comma

File: frontend/test/specs/institutionLinksControllerSpec.js [hunk 7]
```diff
@@ -0,0 +1,122 @@
+'use strict';
+
+(describe('Test InstitutionLinksController', function() {
+
+    var instLinksCtrl, httpBackend, scope, institutionService, createCtrl, state;
+
+    var INSTITUTIONS_URI = "/api/institutions/";
+
+    var parent_institution_test = {
```
Comment (raonismaneoto) [inline frontend/test/specs/institutionLinksControllerSpec.js]: use the javascript pattern, camelCase

File: frontend/test/specs/institutionLinksControllerSpec.js [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+        name: 'parentInst',
+        key: 'parentKey',
+    };
+
+    var children_institutions_test = [
```
Comment (raonismaneoto) [inline frontend/test/specs/institutionLinksControllerSpec.js]: use the javascript pattern, camelCase

File: frontend/test/specs/institutionLinksControllerSpec.js [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+    beforeEach(inject(function($controller, $httpBackend, $rootScope, $state, 
+            InstitutionService) {
+        httpBackend = $httpBackend;
+        scope = $rootScope.$new();
+        state = $state;
+        institutionService = InstitutionService;
+
+        httpBackend.expect('GET', INSTITUTIONS_URI + institution.key).respond(institution);
+        httpBackend.when('GET', 'institution/institution_page.html').respond(200);
+        httpBackend.when('GET', 'institution/removeInstDialog.html').respond(200);
```
Comment (raonismaneoto) [inline frontend/test/specs/institutionLinksControllerSpec.js]: is it necessary?

File: frontend/test/specs/institutionLinksControllerSpec.js [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+        httpBackend.when('GET', "main/main.html").respond(200);
+        httpBackend.when('GET', "home/home.html").respond(200);
```
Comment (raonismaneoto) [inline frontend/test/specs/institutionLinksControllerSpec.js]: same thing

File: frontend/test/specs/institutionLinksControllerSpec.js [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+        it('hasChildrenInst() should return true if has at least one children institution', function() {
+            expect(instLinksCtrl.hasChildrenInst()).toBeTruthy();
+        });
+
+        it('hasChildrenInst() should return false if not has at least one children institution', function() {
+            instLinksCtrl.childrenInstitutions = [];
+            expect(instLinksCtrl.hasChildrenInst()).toBeFalsy();
+        });
+
+        afterEach(function() {
```
Comment (raonismaneoto) [inline frontend/test/specs/institutionLinksControllerSpec.js]: why do you need this?

File: frontend/test/specs/institutionLinksControllerSpec.js [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+            expect(instLinksCtrl.parentInstitution).toEqual(parent_institution_test);
+        });
+
+        it('Should has two children institutions', function() {
+            expect(instLinksCtrl.childrenInstitutions.length).toEqual(2);
+        });
+
+    });
+
+    describe('InstitutionLinksController functions', function() {
```
Comment (raonismaneoto) [inline frontend/test/specs/institutionLinksControllerSpec.js]: it would be better if you split the test in a describe for each function.

**Pull request merged**